trlx/trainer/accelerate_base_trainer.py

```diff
@@ -307,11 +302,33 @@ def save_pretrained(self, directory: Optional[str] = None, **kwargs):
 
     def save(self, directory: Optional[str] = None, **kwargs):
         """Creates a checkpoint of the optimizer, scheduler and model"""
-        self.accelerator.save_state(directory or self.config.train.checkpoint_dir, **kwargs)
+        dst_dir = directory or self.config.train.checkpoint_dir
+        self.accelerator.save_state(dst_dir, **kwargs)
+
+        if self.config.model.peft_config is not None and self.accelerator.is_main_process:
+            # Remove "pytorch_model.bin" because it contains more than necessary,
+            # let save_pretrained recreate it with just the value heads.
+            model_file = os.path.join(dst_dir, "pytorch_model.bin")
+            if os.path.exists(model_file):
+                os.remove(model_file)
+            self.accelerator.unwrap_model(self.model).save_pretrained(dst_dir)
 
     def load(self, directory: Optional[str] = None, **kwargs):
         """Load checkpoint of optimizer, scheduler and a model"""
-        self.accelerator.load_state(directory or self.config.train.checkpoint_dir, **kwargs)
+        if self.config.model.peft_config is not None:
+
+            def load_state_hook(models: List[torch.nn.Module], input_dir: str):
+                with self.accelerator.main_process_first():
+                    for model in models:
+                        model.from_pretrained(input_dir)
+
+            self.accelerator.register_load_state_pre_hook(load_state_hook)
+
+            strict = False
+        else:
+            strict = True
+
+        self.accelerator.load_state(directory or self.config.train.checkpoint_dir, strict=strict, **kwargs)
```

Comment:
For this function I'm not confident that everything I wrote makes sense. I have not tested it in a really distributed setting. And I'm not sure how useful it is because it just sets the internal state of accelerator but it does not set other things like self.model. And when not using peft, testing `trainer.load(checkpoint_dir)` was not working, because of some additional "base_model" prefix in the layer names.